Lib/pyclbr.py

```diff
@@ -164,7 +164,10 @@ def _readmodule(module, path, inpackage=None):
         search_path = path
     else:
         search_path = path + sys.path
-    spec = importlib.util._find_spec_from_path(fullmodule, search_path)
+    try: spec = importlib.util._find_spec_from_path(fullmodule, search_path)
+    except ValueError:
+        # ValueError: __main__.__spec__ is None / is not set
+        spec = None
```

Comment:
It looks like this would just lead to ModuleNotFoundError below, which seems misleading because module does exist. Is ModuleNotFoundError caught somewhere and pyclbr proceeds to next module? Did you test if this change fixes the reported bug?

Reply:
The change from ValueError to an ImportError (subclass) makes it to be processed reasonable (skip to next) in the caller https://github.com/python/cpython/blob/3.10/Lib/pyclbr.py#L243 / `visit_Import()`  in the call stack shown by the trace in bpo 43299 (2 levels higher). And solves the bug in that case at least - and similar in other callers of `_readmodule()`
In that case the module (__main__) can indeed not be found in the filesystem for parsing. regular modules have a spec. Raising an ImportError with extra specific text could be another option.